src/+otp/+kortewegdevries/KortewegdeVriesProblem.m

```diff
@@ -0,0 +1,102 @@
+classdef KortewegdeVriesProblem < otp.Problem
+
+    % Ascher, Uri M. and Robert I. McLachlan. “On Symplectic and 
+    % Multisymplectic Schemes for the KdV Equation.” 
+    % Journal of Scientific Computing 25 (2005): 83-104.
+    %
+    % Robert M. Miura, Clifford S. Gardner, Martin D. Kruskal;
+    % Korteweg‐de Vries Equation and Generalizations. II. 
+    % Existence of Conservation Laws and Constants of Motion. 
+    % J. Math. Phys. 1 August 1968; 9 (8): 1204–1209. 
+    % https://doi.org/10.1063/1.1664701
+
+    methods
+        function obj = KortewegdeVriesProblem(timeSpan, y0, parameters)
+            
+            obj@otp.Problem('Korteweg de Vries Equation', [], ...
+                timeSpan, y0, parameters);
+            
+        end
+    end
+    
+    properties (SetAccess = private)
+        DistanceFunction
+        numInvariants
+        Invariant
+        JacobianInvariant
+    end
+    
+    methods (Static)
+      
+    end
+    
+    methods (Access = protected)
+        function onSettingsChanged(obj)
+                        
+            Xdomain = obj.Parameters.Domain;
+            Nx = obj.Parameters.Nx;
+            Theta = obj.Parameters.Theta;
+            Alpha = obj.Parameters.Alpha;
+            Nu = obj.Parameters.Nu;
+            Rho = obj.Parameters.Rho;
+
+            if obj.NumVars ~= Nx
+                warning('OTP:inconsistentNumVars', ...
+                    'NumVars is %d, but there are %d grid points', ...
+                    obj.NumVars, Nx);
+            end
+            
+            bc = 'C';
+            
+            Dx = otp.utils.pde.Dd(Nx, Xdomain, 1, 1, bc);
+            DxT = Dx.';
+            
+            dx = obj.Parameters.Grid(2) - obj.Parameters.Grid(1);
```

Comment:
Why is the grid a parameter? it can be determined from the domain and the number of elements. Please remove.

Reply:
I need it to compute the initial condition and distances. I dont want to recompute the grid everytime. But I dont mind making this change

Reply:
If you change N, then the grid would ned to be recomputed. that should happen in onSettingsChanged. We don't want to force the user to set N, Y0 AND Grid.

Reply:
That is true. In my input thing, The user is not setting the Grid. IT computed using domain and Nx. I need the grid to make initial conditions. I did try making grid protected and private, but then, the canonical object cannot initialize it. 

One thing I will do is make sure the grid changes when there is a change in domain or Nx.